docs/proposals/user-namespace.md

```diff
@@ -0,0 +1,206 @@
+run those images on the cluster.
+
+[Private user namespaces](http://man7.org/linux/man-pages/man7/user_namespaces.7.html) 
+provide the abiltity to isolate user IDs and group IDs.  This allows
+a process to have different UID and GID values inside and outside of a 
+user namespace.  Ie. running as UID 0 inside the namespace and running
+as a non-root UID outside the namespace so that a process has full 
+privileges inside its isolated namespace but is otherwise unprivileged.
+
+## Container Runtime Support
+
+### Docker
+
+Docker 1.10 supports [user namespace remapping](https://docs.docker.com/v1.10/engine/reference/commandline/daemon/#daemon-user-namespace-options)
+in the docker daemon.  This allows you to run as the root user in a
+container while running as a non-root user in the host.
+
+In Docker, you may provide the `--userns-remap` option to the daemon
+to enable remapping.  This option will read the `/etc/subuid` and
+the `/etc/subgid` files to determine [how to map UIDs and GIDs
+in the container](https://docs.docker.com/engine/reference/commandline/dockerd/#detailed-information-on-subuidsubgid-ranges).
+
+Additionally, the `HostConfig` for a container now contains a parameter
+that lets you escape the private user namespace and indicate that you
+would like to [use the host's user namespace](https://github.com/docker/engine-api/blob/4290f40c056686fcaa5c9caf02eac1dde9315adf/types/container/host_config.go#L310).
+By using this setting, administrators may run containers outside of the user
+namespace while still requiring other containers to run as a remapped
+user.
+
+### rkt
+
+[rkt supports user namespaces](https://coreos.com/rkt/docs/latest/devel/user-namespaces.html)
+with a systemd-nspawn based implementation.  Each pod started will
+be allocated a random UID range. 
+
+To use private user namespaces in rkt you specify the `--private-users`
+flag on the rkt run command.
+
+
+## Kubernetes Support
+
+As of 1.5 [experimental support](https://github.com/kubernetes/kubernetes/pull/31169)
+has been added which will default the user namespace to host when 
+a pod is using features that require host namespaces (see defaulting scenarios
+described in the design below).  This proposal is 
+building on the experimental work as phase 2 experimental support with 
+the goal of fixing the current known issues.
+
+### Known Issues
+
+These are the known issues (as of writing) that will be hit when
+enabling user namespace remapping in a Kubernetes cluster.
+
+1. Negative oomScoreAdj values fail.  In this case the pause container
+cannot be started and shows an error `write /proc/self/oom_score_adj: permission denied`
+in the daemon log.  Failing in docker 1.10, working in 1.11+
+1. Cannot share the host or a container's network namespace when user
+namespaces are enabled.  [Fixed in 1.11](https://github.com/docker/docker/pull/21383).
+1. Permission denied mounting.  Occurs due to ownership permissions
```

Comment:
If we just use the FSGroup for this, does it work?  I actually think this represents a bug...